src/main/sensors/compass.c

```diff
@@ -460,42 +460,56 @@ uint32_t compassUpdate(timeUs_t currentTimeUs)
         alignSensorViaRotation(mag.magADC, magDev.magAlignment);
     }
 
+    // get stored cal/bias values
     flightDynamicsTrims_t *magZero = &compassConfigMutable()->magZero;
 
-    if (tCal != 0) {
-        if (cmpTimeUs(tCal, currentTimeUs) > 0) {
-            LED0_TOGGLE;
-
-            // it is assumed that the user has started to move the quad if squared norm of rotational speed vector is greater than GYRO_NORM_SQUARED_MIN
+    // ** perform calibration, if initiated by switch or Configurator button **
+    if (magCalProcessActive) {
+        if (cmpTimeUs(magCalEndTime, currentTimeUs) > 0) {
+                // compare squared norm of rotation rate to GYRO_NORM_SQUARED_MIN
             float gyroNormSquared = 0.0f;
             for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
                 gyroNormSquared += sq(DEGREES_TO_RADIANS(gyroGetFilteredDownsampled(axis)));
             }
-
-            // check if movement has started
             if (!didMovementStart && gyroNormSquared > GYRO_NORM_SQUARED_MIN) {
+                // movement has started
+                beeper(BEEPER_READY_BEEP); // Beep to alert user to start moving the quad (does this work?)
+                // zero the old cal/bias values
+                for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+                    magZero->raw[axis] = 0;
+                }
                 didMovementStart = true;
-                // starting now, the user has CALIBRATION_TIME_US to move the quad in a figure of eight in all directions
-                tCal = micros() + CALIBRATION_TIME_US;
+                // the user has CALIBRATION_TIME_US from now to move the quad in all directions
+                magCalEndTime = micros() + CALIBRATION_TIME_US;
             }
-
-            // only start calibration after the user has started to move the quad
+            // start acquiring mag data and computing new cal factors
             if (didMovementStart) {
+                // LED will flash at task rate while calibrating, looks like 'ON' all the time.
```

Comment:
isn't it better to use LED_ON then? Or divide update rate so that blinking is visible
```
static int cnt;
if ((++cnt) % 10 == 0) LED0_TOGGLE;
```

Or just:
```
if (millis() % 100 < 50) { LED_ON; } else {LED_OFF};

Reply:
Thanks - `LED0_ON` works great!  For some reason it didn't, before.